src/issues/shareProviders.ts

```diff
@@ -194,9 +202,49 @@ export class GitHubPermalinkShareProvider extends AbstractShareProvider implemen
 	}
 }
 
+export class GitHubPermalinkAsMarkdownShareProvider extends GitHubPermalinkShareProvider {
+
+	constructor(repositoryManager: RepositoriesManager, gitApi: GitApiImpl) {
+		super(repositoryManager, gitApi, 'githubComPermalinkAsMarkdown', vscode.l10n.t('Copy GitHub Permalink as Markdown'), 12);
+	}
+
+	async provideShare(item: vscode.ShareableItem): Promise<vscode.Uri | string | undefined> {
+		const link = await super.provideShare(item);
+
+		const text = await this.getMarkdownLinkText(item);
+		if (link) {
+			return `[${text?.trim() ?? ''}](${link.toString()})`;
+		}
+	}
+
+	private async getMarkdownLinkText(item: vscode.ShareableItem): Promise<string | undefined> {
+		const fileName = pathLib.basename(item.resourceUri.path);
+
+		if (item.selection) {
+			const document = await vscode.workspace.openTextDocument(item.resourceUri);
+
+			const editorSelection = item.selection.start === item.selection.end
+				? item.selection
+				: new vscode.Range(item.selection.start, new vscode.Position(item.selection.start.line + 1, 0));
+
+			const selectedText = document.getText(editorSelection);
+			if (selectedText) {
+				return selectedText;
+			}
+
+			const wordRange = document.getWordRangeAtPosition(item.selection.start);
+			if (wordRange) {
+				return document.getText(wordRange);
+			}
+		}
+
+		return fileName;
```

Comment:
This implementation should address https://github.com/microsoft/vscode-pull-request-github/issues/4663 once we inline Share (which requires async context menus in Electron)